engine/profiler/src/profile_render.cpp

```diff
@@ -641,35 +645,31 @@ namespace dmProfileRender
         if (!current_frame)
             return;
 
-        ProfilerThread* last_thread = GetSelectedThread(render_profile, render_profile->m_CurrentFrame);
-
         render_profile->m_CurrentFrame = DuplicateProfilerFrame(current_frame);
 
         ProfilerThread* current_thread = GetSelectedThread(render_profile, render_profile->m_CurrentFrame);
-        if (last_thread->m_NameHash != current_thread->m_NameHash)
+        if (last_thread_name != current_thread->m_NameHash)
         {
-            last_thread = 0;
             render_profile->m_MaxFrameTime = 0;
         }
 
-        //uint64_t last_frame_time = last_thread ? last_thread->m_SamplesTotalTime : 0;
         uint64_t this_frame_time = current_thread->m_SamplesTotalTime;
 
         bool new_peak_frame = render_profile->m_MaxFrameTime < this_frame_time;
         render_profile->m_MaxFrameTime = dmMath::Max(render_profile->m_MaxFrameTime, this_frame_time);
 
-        if (render_profile->m_Mode == PROFILER_MODE_SHOW_PEAK_FRAME)
+        if (new_peak_frame)
```

Comment:
Move those condition outside of `if (render_profile->m_Mode == PROFILER_MODE_SHOW_PEAK_FRAME)` because profiler updates `m_MaxFrameTime` every frame but peak frame stores only when special profiler mode is active. It can lead to situation when peak frame happened in the past before switching to `PROFILER_MODE_SHOW_PEAK_FRAME`. As result peak frame never stored and profiler is continue showing current frame as active.